From 56864bb3c0fe4db9534401a0cfba272b49d425f7 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 23 Oct 2020 21:13:23 +0300
Subject: [PATCH] [Sw-20154] remove TODO in preboot/uboot load address

This patch removes the TODO for reading load address
of preboot and uboot from specs instead of hard
coding the same. This is needed for Gaudi2

Change-Id: Ibadce07c6d853035c489fa1c22df838dbab319cc
Signed-off-by: Pardha Kesapragada<pkesapragada@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-spi.sh b/board/habanalabs/common/post-image-spi.sh
index b642c67..4d49814 100755
--- a/board/habanalabs/common/post-image-spi.sh
+++ b/board/habanalabs/common/post-image-spi.sh
@@ -11,6 +11,10 @@
 UBOOT_CMDFILE=${UBOOT_CMDFILE:-${BOARD_BASE}/uboot-eeprom.cmd}
 UBOOT_FITFILE=u-boot.itb
 UBOOT_BINFILE=${UBOOT_BINFILE:-${UBOOT_FITFILE}}
+# Set the Preboot and Uboot text bases
+HL_FW_IF_FILE=$SPECS_EMBEDDED_ROOT/${HL_DEV}/${HL_DEV}_fw_if.h
+PREBOOT_TEXT_BASE=`grep PREBOOT_TEXT_BASE $HL_FW_IF_FILE | sed 's/^.*\(0x.*\)$/\1/'`
+UBOOT_TEXT_BASE=`grep UBOOT_TEXT_BASE $HL_FW_IF_FILE | sed 's/^.*\(0x.*\)$/\1/'`
 
 source ${COMMON_BASE}/image-utility.sh
 
@@ -39,11 +43,6 @@
 
 function get_image_loadaddr {
 	local idx=$1
-	# TODO: rework the below to recieve the addresses from specs instead
-	#	of hard coding them here
-	local PREBOOT_TEXT_BASE=0x7FF0004000
-	local UBOOT_TEXT_BASE=0x7FF0100000
-
 	case $idx in
 	1) # Preboot
 		echo $PREBOOT_TEXT_BASE
